CIRC-6581 Initial drop of documentation covering CUA (#191)
circonus/docs: content/circonus/agents/circonus-unified-agent/basic_usage/docker.md

---
title: Docker
weight: 10
---

# CUA for Docker

This page outlines the installation and configuration of CUA for Docker.

## Installation

> 1. Create a configuration file (e.g. use example from the [repository](https://github.com/circonus-labs/circonus-unified-agent/tree/master/etc/example-circonus-unified-agent.conf))

Reviewer: ```suggestion
1. Create a configuration file from the [repository example](https://github.com/circonus-labs/circonus-unified-agent/tree/master/etc/example-circonus-unified-agent.conf)
```

2. In `outputs.circonus`:

>     a. Set `api_token`

Reviewer: ```suggestion
    a. Set `api_token` to your [Circonus api token](/circonus/administration/api-tokens/)
```

>     b. Set `check_name_prefix` to ensure the check target can be found again when container is restarted - agent will container's "hostname" as the check target by default.

Reviewer: ```suggestion
    b. Set `check_name_prefix` to ensure the check target can be found again after a container is restarted - agent will use the container's "hostname" as the check target by default.
```

Use one of the following commands:

```sh
docker run -d --name=circonus-unified-agent \
    -v $PWD/circonus-unified-agent.conf:/etc/circonus-unified-agent/circonus-unified-agent.conf:ro \
    circonus/circonus-unified-agent
```

or

```sh
docker run -d --name=circonus-unified-agent \
    --mount type=bind,src=$PWD/circonus-unified-agent.conf,dst=/etc/circonus-unified-agent/circonus-unified-agent.conf \
    circonus/circonus-unified-agent
```

## Integrations

> In some cases, you may wish to collect HOST metrics from within the container. To do so, use one of the following commands:

Reviewer: Why is HOST in all caps?

Author: That came from Matt, I'm fine either way.
